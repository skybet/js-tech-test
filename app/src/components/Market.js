import React from "react";
import Outcome from "./Outcome";
import { useMarket } from "../hooks/";

const Market = ({ id }) => {
  const market = useMarket(id);

  if (!market) {
    return <div>Loading...</div>;
  }

  const { name, outcomes } = market;

  return (
    <div className="c-market">
      <div className="c-market__title">{name}</div>
      <div>
        {outcomes &&
          outcomes.map(outcome => <Outcome key={outcome} id={outcome} />)}
      </div>
    </div>
  );
};

export default Market;
