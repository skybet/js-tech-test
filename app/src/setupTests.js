// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render } from "@testing-library/react";
import { OddsContext } from "./App";

jest.mock("./hooks/useMarket", () =>
  jest.fn(() => ({
    marketId: 111,
    name: "Both teams to score",
    outcomes: [111, 222]
  }))
);

jest.mock("./hooks/useOutcome", () =>
  jest.fn(() => ({
    name: "Yes",
    price: {
      num: 3,
      den: 1,
      decimal: 4
    }
  }))
);

jest.mock("./hooks/useLiveEvents", () =>
  jest.fn(() => [
    {
      eventId: 111,
      name: "Everton vs West Ham",
      markets: [111, 222],
      competitors: [{ name: "Everton" }, { name: "West Ham" }],
      scores: { home: 3, away: 0 }
    }
  ])
);

export const renderWithContext = (component, oddsType = "fractional") =>
  render(
    <OddsContext.Provider value={[oddsType, jest.fn()]}>
      {component}
    </OddsContext.Provider>
  );
