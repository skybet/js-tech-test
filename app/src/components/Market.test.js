import React from "react";
import Market from "./Market";
import { renderWithContext } from "../setupTests";

describe("<Market />", () => {
  test("renders as expected", async () => {
    const { asFragment } = renderWithContext(<Market />);
    expect(asFragment()).toMatchSnapshot();
  });
});
