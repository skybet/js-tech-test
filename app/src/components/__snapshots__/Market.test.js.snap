// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Market /> renders as expected 1`] = `
<DocumentFragment>
  <div
    class="c-market"
  >
    <div
      class="c-market__title"
    >
      Both teams to score
    </div>
    <div>
      <div
        class="c-outcome"
      >
        <span
          class="u-clr--blue u-mr--100"
        >
          Yes
        </span>
        <span
          class="u-clr--red"
        >
          3/1
        </span>
      </div>
      <div
        class="c-outcome"
      >
        <span
          class="u-clr--blue u-mr--100"
        >
          Yes
        </span>
        <span
          class="u-clr--red"
        >
          3/1
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
